### PR TITLE
Handle null statistics in Iceberg Dynamic Filtering

### PR DESCRIPTION
**Pull request by alexjo2144**
Fixes https://github.com/trinodb/trino/issues/9937

### CODE REVIEW

**Review by jirassimok — APPROVED** [review]
Looks good to me.

Maybe the test could use some explanatory comments, but so could all the other tests in that file.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java [hunk 1]
```diff
@@ -231,7 +239,11 @@ static boolean fileMatchesPredicate(
         return true;
     }
 
-    private static Domain domainForStatistics(io.trino.spi.type.Type type, Object lowerBound, Object upperBound, boolean containsNulls)
+    private static Domain domainForStatistics(
+            io.trino.spi.type.Type type,
+            @Nullable Object lowerBound,
+            @Nullable Object upperBound,
+            boolean containsNulls)
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java:246]: nit: preexiting - `mayContainNulls` or `nullsAllowed` would be better name for paramaeter.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSplitSource.java [hunk 2]
```diff
@@ -284,4 +284,54 @@ public void testBigintStatisticsPruning()
                 upperBound,
                 ImmutableMap.of(1, 1L)));
     }
+
+    @Test
+    public void testNullHandling()
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSplitSource.java:289]: Maybe call the method `testNullStatisticsMaps`

**Comment by alexjo2144** [comment]
Renamed both, thanks Lukasz

**Pull request merged**